pymatgen/io/openmm/inputs.py

```diff
@@ -0,0 +1,635 @@
+
+    @classmethod
+    def from_string(cls, contents: str):
+        """
+        This is a template that should be overwritten. Replace XmlInput with child class.
+        """
+        return XmlInput(XmlSerializer.deserialize(contents))
+
+
+class SystemInput(XmlInput):
+    @classmethod
+    def from_string(cls, contents: str):
+        return SystemInput(XmlSerializer.deserialize(contents))
+
+    def get_system(self) -> System:
+        return XmlSerializer.deserialize(self.content)
+
+
+class IntegratorInput(XmlInput):
+    @classmethod
+    def from_string(cls, contents: str):
+        return IntegratorInput(XmlSerializer.deserialize(contents))
+
+    def get_integrator(self) -> Integrator:
+        return XmlSerializer.deserialize(self.content)
+
+
+class StateInput(XmlInput):
+    @classmethod
+    def from_string(cls, contents: str):
+        return StateInput(XmlSerializer.deserialize(contents))
+
+    def get_state(self) -> State:
+        return XmlSerializer.deserialize(self.content)
+
+
+class OpenMMSet(InputSet):
+    @classmethod
+    def from_directory(
+        cls,
+        directory: Union[str, Path],
+        topology_file: str = "topology.pdb",
+        system_file: str = "system.xml",
+        integrator_file: str = "integrator.xml",
+        state_file: str = "state.xml",
+    ):
+        """
+        An InputSet for OpenMM Simulations.
+
+        This is a container for the serialized topology, system, integrator, and state files.
+        Topology is stored as a pdb and the other files are stored as xml. The state file is
+        optional but positions must be manually assigned if no state is provided.
+
+        Args:
+            directory: directory that holds the input files.
+            topology_file: name of the pdb file with topological information.
+            system_file: name of the serialized System xml file.
+            integrator_file: name of the serialized Integrator xml file.
+            state_file: name of the serialized State xml file. If there is no state_file,
+                then positions must be set for simulation.
+
+        Returns:
+            an OpenMMSet
+        """
+        dir = Path(directory)
+        topology_input = TopologyInput.from_file(dir / topology_file)
+        system_input = SystemInput.from_file(dir / system_file)
+        integrator_input = IntegratorInput.from_file(dir / integrator_file)
+        inputs = {
+            topology_file: topology_input,
+            system_file: system_input,
+            integrator_file: integrator_input,
+        }
+        openmm_set = OpenMMSet(
+            inputs=inputs,
+            topology_file=topology_file,
+            system_file=system_file,
+            integrator_file=integrator_file,
+        )
+        if Path(dir / state_file).is_file():
+            openmm_set.inputs[state_file] = StateInput.from_file(dir / state_file)
+            openmm_set.state_file = state_file  # should this be a dict-like assignment?
```

Comment:
No, this is OK as is. For `InputSet` dict-like assignment will populate the `.inputs` dictionary rather than the attributes. So in the line above, you could also say `openmm_set[state_file] = ...` and have the same effect.

`openmm_set.state_file = ...` should work as expected, as you can set and get arbitrary attributes on an `InputSet`.

Reply:
Great. Thanks for the clarification.